src/dos/dos.cpp

```diff
@@ -1204,6 +1204,43 @@ static Bitu DOS_26Handler(void) {
     return CBRET_NONE;
 }
 
+bool set_ver(const char *word, const char *args)
```

Comment:
*(just a suggestion, feel free to investigate, but it's not a blocker in any way)*

I believe implementation of this function could be greatly simplified by using something like:
```
const int tokens = sscanf(args, "%d.%c%c%c", &major, &minor[0], &minor[1], &minor[2]);
```
Especially if this function will return `DOS_Version` struct and return early.

Reply:
It should also work, but the function will do some checking, and separate the "VER set version_number" and "VER set major_version [minor_version]" cases, so I doubt there will be much simplification in the end compared to the current code.